Address Masonry First Load Image Failures

stwrt opened the pull request:
#### 605d174ea587fb7635afe58f0fbb0f79ee3df6b6
<pre>
Address Masonry First Load Image Failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=251733">https://bugs.webkit.org/show_bug.cgi?id=251733</a>

Reviewed by Matt Woodrow.

Masonry has been having some interesting behavior with pure images
due to heights often being incorrectly rendered as 0px. This patch
fixes the first load problem. This seems to be connected to not properly
populating grid positions.

* Source/WebCore/rendering/GridMasonryLayout.cpp:
(WebCore::GridMasonryLayout::performMasonryPlacement):

Canonical link: <a href="https://commits.webkit.org/259978@main">https://commits.webkit.org/259978@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f402565420bc8ce7cd778379bd0c6d1dbb59640

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106170 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15221 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39003 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115352 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115038 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110074 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16663 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6406 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98373 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111928 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12680 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95649 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40213 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94553 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27292 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81898 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8467 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28644 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8967 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5199 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14583 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48190 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6900 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10503 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mattwoodrow commented on Source/WebCore/rendering/GridMasonryLayout.cpp:
```diff
@@ -43,7 +43,10 @@ void GridMasonryLayout::performMasonryPlacement(unsigned gridAxisTracks, GridTra
+    resizeAndResetRunningPositions();
+
+    m_renderGrid.populateGridPositionsForDirection(ForColumns);
+    m_renderGrid.populateGridPositionsForDirection(ForRows);
```
I think we only need to do this for the grid-axis direction, since that's the one we did sizing for. The masonry axis doesn't really have tracks, and we haven't done any sort of 'sizing', so copying across positions for it is a bit meaningless.

stwrt replied:
I probably want some code here to track which direction is the masonry axis.

stwrt committed:
Address Masonry First Load Image Failures
https://bugs.webkit.org/show_bug.cgi?id=251733

Reviewed by Matt Woodrow.

Masonry has been having some interesting behavior with pure images
due to heights often being incorrectly rendered as 0px. This patch
fixes the first load problem. This seems to be connected to not properly
populating grid positions.

* Source/WebCore/rendering/GridMasonryLayout.cpp:
(WebCore::GridMasonryLayout::performMasonryPlacement):

Canonical link: https://commits.webkit.org/259978@main
